src/events/interactionCreate.js

```diff
@@ -0,0 +1,58 @@
+import chalk from 'chalk';
+import { sanitize, validateMessage, registerComponents } from 'utils/discord';
+
+/**
+ * Handles interaction events.
+ */
+const InteractionEvent = {
+  name: 'interactionCreate',
+  async execute(client, interaction) {
+    try {
+      switch (interaction.type) {
+        // Command interactions
+        case 'APPLICATION_COMMAND': {
+          const { commandName, options } = interaction;
+
+          const command = client.commands.get(commandName);
+          if (!command) return;
+
+          const output = await command.execute({
+            ...client,
+            options: options?.data?.map(({ value }) => sanitize(value)),
+          });
+          if (!output) return;
+
+          const data = validateMessage(output);
+          await interaction.reply(data);
+          if (!data.components) return;
+
+          const message = await client.api
+            .webhooks(client.user.id, interaction.token)
+            .messages('@original')
+            .get();
```

Comment:
Would prefer if this was native or if there was a better way of keeping track of parent interactions/callbacks for buttons.

Reply:
I think [`Interaction.reply`](https://discord.js.org/#/docs/main/stable/class/CommandInteraction?scrollTo=reply) will return something, if it doesn't then we can use [`Interaction.fetchReply`](https://discord.js.org/#/docs/main/stable/class/CommandInteraction?scrollTo=fetchReply)

Reply:
Maybe, cache the command interaction and get the interaction when we get a button interaction